src/System.Private.CoreLib/src/System/EETypePtr.cs

```diff
@@ -385,22 +384,54 @@ internal uint ValueTypeSize
             }
         }
 
-        internal RuntimeImports.RhCorElementType CorElementType
+        internal CorElementType CorElementType
```

Comment:
Is this used on any hot paths? This change makes this quite a bit slower.

Reply:
It's used mostly in shared code around enums and arrays. They mostly just care about primitive types, so we could make this faster if we forgo correctness for the other values. We were already not reporting ARRAY/SZARRAY/STRING that Array.cs tests for to make any optimization anyway.

Then it's used in the widening code that I didn't want to touch because it's another scary change (offsetting into arrays). That should be a separately reviewed thing, ideally. We can add it as a commit in this pull request later, but I don't want it to distract from the main change here.